doc/usage/extensions/autosummary.rst

```diff
@@ -201,6 +201,25 @@ also use these config values:
 
    .. versionadded:: 2.1
 
+   .. versionchanged:: 4.3
+
+      If ``autosummary_ignore___all__`` is ``False``, this configuration value
+      is ignored for members listed in ``__all__``.
+
+.. confval:: autosummary_ignore___all__
+
+   If ``False`` and a module has the ``__all__`` attribute set, autosummary
+   documents every member listed in ``__all__`` and no others. Default is
+   ``True``
```

Comment:
Please disable this by default at this moment. I agree this must be useful. But I'd not like to change the behavior of autosummary during 4.x to keep compatibility. So it should be changed since 5.0.

Reply:
The new behaviour is disabled by default - ie, we ignore `__all__` by default, so `automodule_ignore_module_all` is `True`. If this is too confusing I can flip it so it's `automodule_respect_module_all`, but I thought this way around would make more sense once the default is changed in 5.0